### PR TITLE
[vSphere] Move vCenter credentials to Cluster provider config

### LINKED ISSUE #227: [vSphere] Move vCenter credentials to Cluster provider config
For de-duplication

/assign @karan

### PR DESCRIPTION
**Pull request by karan**
**What this PR does / why we need it**:

1. Split vsphere provider config into one for machine and one for cluster.
2. Move the user, password, server to cluster provider config.

Validated by creating a cluster using `vsphere-deployer`.

**Which issue(s) this PR fixes**:
Fixes #227

### CODE REVIEW

File: vsphere-deployer/cluster.yaml.template [hunk 1]
```diff
@@ -12,4 +12,7 @@ spec:
     providerConfig:
       value:
         apiVersion: "vsphereproviderconfig/v1alpha1"
-        kind: "VsphereProviderConfig"
\ No newline at end of file
+        kind: "VsphereClusterProviderConfig"
```
Comment (mkjelland) [inline vsphere-deployer/cluster.yaml.template]: The [README](https://github.com/karan/cluster-api/tree/73086ded2e4fbb514aeee4edc7dd238469dd63f9/vsphere-deployer) for vsphere-deployer will need to be updated to include instructions on creating the cluster.yaml file.

Comment (karan) [reply vsphere-deployer/cluster.yaml.template]: Done

**Comment by karan** [comment]
Sure.

/assign @roberthbailey

**Pull request merged**